src/effects/effectchainslot.h

```diff
@@ -53,30 +58,77 @@ class EffectChainSlot : public QObject {
     QDomElement toXml(QDomDocument* doc) const;
     void loadChainSlotFromXml(const QDomElement& effectChainElement);
 
-  signals:
-    // Indicates that the effect pEffect has been loaded into slotNumber of
-    // EffectChainSlot chainNumber. pEffect may be an invalid pointer, which
-    // indicates that a previously loaded effect was removed from the slot.
-    void effectLoaded(EffectPointer pEffect, unsigned int chainNumber,
-                      unsigned int slotNumber);
 
+    // Activates EffectChain processing for the provided channel.
+    // TODO(Kshitij) : Make this function private once EffectRack layer is removed
+    void enableForInputChannel(const ChannelHandleAndGroup& handle_group);
+    const QSet<ChannelHandleAndGroup>& enabledChannels() const;
+    void disableForInputChannel(const ChannelHandleAndGroup& handle_group);
+
+    // TODO(Kshitij) : Make this function private once EffectRack layer is removed
+    void updateEngineState();
+
+    // Get the human-readable name of the EffectChain
+    const QString& name() const;
+    void setName(const QString& name);
+
+    // Get the human-readable description of the EffectChain
+    QString description() const;
+    void setDescription(const QString& description);
+
+    // TODO(Kshitij) : Make this function private once EffectRack layer is removed
```

Comment:
This function should be removed when the EffectRack layer is removed, not made private.